tracer/src/Datadog.Trace/AppSec/Security.cs

```diff
@@ -32,14 +30,18 @@ internal class Security : IDatadogSecurity, IDisposable
         private static Security _instance;
         private static bool _globalInstanceInitialized;
         private static object _globalInstanceLock = new();
-
+        private readonly Concurrency.ReaderWriterLock _wafLocker = new();
         private readonly SecuritySettings _settings;
+        private LibraryInitializationResult _libraryInitializationResult;
+
+        // todo: make it IWaf? when enable nullable. Having security on doesn't mean waf has init properly
```

Comment:
don't really understand this comment and probably better to remove the todo and make a decision one way of the other.

Reply:
removed in 9daf1e931a478a25f20f4c693fc374289724d86d